extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/JsonWebKeyCache.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.oidc.runtime;
+
+import java.security.Key;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jose4j.jwk.JsonWebKey;
+import org.jose4j.jwk.JsonWebKeySet;
+import org.jose4j.lang.JoseException;
+
+import io.quarkus.oidc.OIDCException;
+
+public class JsonWebKeyCache {
+    private static final String RSA_KEY_TYPE = "RSA";
+    private static final String EC_KEY_TYPE = "EC";
+    private static final String SIGNATURE_USE = "sig";
+
+    private Map<String, Key> keys = new HashMap<>();
```

Comment:
Aren't you afraid of an out-of-memory issue? 
Nobody removes anything from the map.

Reply:
@cescoffier Hi Clement - good question and I was  not sure immediately what was the right answer :-). It is safe - this map never grows.
We get this Map on the start up - it represents a JWK set - which is typically a single key in providers like Keycloak, Auth0, etc, a few max. It never grows - and when Keycloak rotates the keys and Quarkus fetches a new map - this map is replaced as opposed to augmented, please see [here](https://github.com/quarkusio/quarkus/pull/14953/files#diff-c5749413dbc9d20dc1c0e93ed83597f458c909f60eff7ca7b1ad641fe6edc1f9R190).
Thanks for spotting this Map :-), it is worth checking indeed